### PR TITLE
Tor socks proxy support

### PR DESCRIPTION
**Pull request by shavit**
To use the proxy use the `Tor(url *url.URL) (err error)` method

````
var err error
bow := surf.NewBrowser()
//err:= bow.Open("http://check.torproject.org")
//  if err != nil {
//    panic(err)
//  }
// Without proxy: Sorry. You are not using Tor.
//fmt.Println(bow.Title())

// With proxy: Congratulations. This browser is configured to use Tor.
u, _ := url.Parse("socks5://127.0.0.1:9050")
bow.Tor(u)
err = bow.Open("http://check.torproject.org")
  if err != nil {
   panic(err)
 }
fmt.Println(bow.Title())
````

### CODE REVIEW

**Comment by lxt2** [comment]
I'm reluctant to add a Tor specific method when there is already possible via `SetTransport`. Perhaps this could be a `SetProxy` instead?

**Comment by shavit** [comment]
Yes, see https://github.com/headzoo/surf/pull/56/commits/5948ff4af84a1e94b415912849b696c8f8526a52

**Comment by headzoo** [comment]
My only reservation is passing a `*url.URL` to `SetProxy()` instead of a string like the other methods which take a url.

**Comment by shavit** [comment]
ok, see https://github.com/headzoo/surf/pull/56/commits/06b521e6b9a8326064e6304f33fbcb8ede384e20

**Comment by headzoo** [comment]
Thanks, looks good to me. I'll wait on @lxt2 so we can merge a bunch of these pull requests at once into one patch release. Should be soon.

File: README.md [hunk 1]
```diff
@@ -6,12 +6,17 @@ Surf
...
 [![MIT License](https://img.shields.io/badge/license-MIT-blue.svg?style=flat-square)](https://raw.githubusercontent.com/headzoo/surf/master/LICENSE.md)
 
-Surf is a Go (golang) library that implements a virtual web browser that you control pragmatically.
+Surf is a Go (golang) library that implements a virtual web browser that you control programmatically.
 Surf isn't just another Go solution for downloading content from the web. Surf is designed to behave
 like web browser, and includes: cookie management, history, bookmarking, user agent spoofing
 (with a nifty user agent builder), submitting forms, DOM selection and traversal via jQuery style
 CSS selectors, scraping assets like images, stylesheets, and other features.
 
+**NOTE: DEVELOPMENT ON THIS PROJECT IS VERY SLOW!**
```
Comment (lxt2) [inline README.md]: Hrm, not sure why this is being included.

Comment (shavit) [inline README.md]: It was an old pull request

**Comment by lxt2** [comment]
Moving to #72.

**Pull request closed without merging**